### PR TITLE
Send email when onboarding PG using ms-notification-manager

### PR DESCRIPTION
**Pull request by empassaro**
#### List of Changes

Moved all the logic of email sending which relies on selfcare-ms-notification-manager from dashboard-backend module to ms-core module for the following APIs:
	POST onboarding/operators
	POST onboarding/subdelegates
Added logic to send confirmation email when onboarding pg calls selfcare-ms-notification-manager's API

#### Motivation and Context

With this update we can send confirmation emails when onboarding PG and when operators and subdelegates are onboarded

#### How Has This Been Tested?

Tested in local environment, calling the APIs described to check if they persist the data and send the emails.

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by manuraf** [comment]
Title and description don't follow our convention template (https://github.com/pagopa/selfcare-ms-core/blob/main/.github/PULL_REQUEST_TEMPLATE.md). For example, look at https://github.com/pagopa/selfcare-ms-core/pull/144

**Review by manuraf — CHANGES_REQUESTED** [review]
Title and description don't follow our convention template (https://github.com/pagopa/selfcare-ms-core/blob/main/.github/PULL_REQUEST_TEMPLATE.md). For example, look at https://github.com/pagopa/selfcare-ms-core/pull/144

File: core/src/main/java/it/pagopa/selfcare/mscore/core/UserRelationshipServiceImpl.java
Comment (empassaro) [inline core/src/main/java/it/pagopa/selfcare/mscore/core/UserRelationshipServiceImpl.java:1]: @KevinSi96 Ho effettuato un merge dei tuoi sviluppi con i nostri. Quando hai un attimo puoi controllare sia tutto ok anche per te?

File: connector-api/src/test/java/it/pagopa/selfcare/mscore/exception/ResourceForbiddenExceptionTest.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class ResourceForbiddenExceptionTest {
+
+    @Test
+    void testConstructor() {
+        assertEquals("Code", (new ResourceForbiddenException("An error occurred", "Code")).getCode());
+    }
+}
+
```
Comment (manuraf) [inline connector-api/src/test/java/it/pagopa/selfcare/mscore/exception/ResourceForbiddenExceptionTest.java:14]: we could avoid to test the exception classess

File: connector/dao/src/test/java/it/pagopa/selfcare/mscore/connector/dao/InstitutionConnectorImplTest.java [hunk 2]
```diff
@@ -58,6 +58,94 @@ class InstitutionConnectorImplTest {
...
+        institutionUpdate.setZipCode("21654");
+
+        Onboarding onboarding = new Onboarding();
+        onboarding.setBilling(new Billing());
+        onboarding.setContract("contract");
+        onboarding.setStatus(RelationshipState.ACTIVE);
+        onboarding.setCreatedAt(OffsetDateTime.now());
+        onboarding.setProductId("productId");
+        onboarding.setUpdatedAt(OffsetDateTime.now());
+        onboarding.setPricingPlan("pricingPal");
```
Comment (manuraf) [inline connector/dao/src/test/java/it/pagopa/selfcare/mscore/connector/dao/InstitutionConnectorImplTest.java:99]: we could avoid to replicate the same definition of an object for each test method....you might implement an utility test (es. createDummy PaymentServiceProvider) that create a dummy object with default value for each attribute and using that when you need a dummy object for test.....look at https://github.com/pagopa/selfcare-ms-core/blob/main/web/src/test/java/it/pagopa/selfcare/mscore/web/TestUtils.java

File: connector/dao/src/test/java/it/pagopa/selfcare/mscore/connector/dao/InstitutionConnectorImplTest.java [hunk 3]
```diff
@@ -58,6 +58,94 @@ class InstitutionConnectorImplTest {
...
+        institutionEntity.setOrigin(Origin.MOCK);
+        institutionEntity.setOriginId("42");
+        institutionEntity.setPaymentServiceProvider(paymentServiceProviderEntity);
+        institutionEntity.setRea("Rea");
+        institutionEntity.setShareCapital("Share Capital");
+        institutionEntity.setSupportEmail("jane.doe@example.org");
+        institutionEntity.setSupportPhone("6625550144");
+        institutionEntity.setTaxCode("Tax Code");
+        institutionEntity.setUpdatedAt(null);
+        institutionEntity.setZipCode("21654");
```
Comment (manuraf) [inline connector/dao/src/test/java/it/pagopa/selfcare/mscore/connector/dao/InstitutionConnectorImplTest.java:142]: ditto

File: core/src/test/java/it/pagopa/selfcare/mscore/core/UserNotificationServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+        labels.add("label");
+
+        User user1 = new User();
+        user1.setId("1");
+        user1.setFiscalCode("ABC123XYZ");
+        CertifiedField<String> nome = new CertifiedField<>();
+        nome.setValue("nome");
+        user1.setName(nome);
+        user1.setFamilyName(nome);
+        user1.setEmail(nome);
```
Comment (manuraf) [inline core/src/test/java/it/pagopa/selfcare/mscore/core/UserNotificationServiceImplTest.java:68]: ditto

**Review by manuraf — CHANGES_REQUESTED** [review]

File: connector-api/src/test/java/it/pagopa/selfcare/mscore/exception/ResourceForbiddenExceptionTest.java (hunk 1 above)
Comment (manuraf) [reply connector-api/src/test/java/it/pagopa/selfcare/mscore/exception/ResourceForbiddenExceptionTest.java:14]: you might remove only this

**Review by andrea-putzu — APPROVED** [review]

**Review by manuraf — APPROVED** [review]

**Pull request merged**